naga/src/back/msl/writer.rs

```diff
@@ -6403,36 +6402,60 @@ template <typename A>
                             .expect("Should have generated this unpacking function earlier.");
                         let func_name = &func.name;
 
-                        write!(self.out, "{}{attribute_name} = ", back::Level(2),)?;
-
                         // Check dimensionality of the attribute compared to the unpacking
                         // function. If attribute dimension is < unpack dimension, then
                         // we need to explicitly cast down the result. Otherwise, if attribute
                         // dimension > unpack dimension, we have to pad out the unpack value
                         // from a vec4(0, 0, 0, 1) of matching scalar type.
 
-                        let needs_truncate_or_padding = am.dimension != func.dimension;
-                        if needs_truncate_or_padding {
+                        let needs_padding_or_truncation = am.dimension.cmp(&func.dimension);
+
+                        if needs_padding_or_truncation != Ordering::Equal {
+                            writeln!(
+                                self.out,
+                                "{}// {attribute_ty_name} <- {:?}",
```

Comment:
Since the swizzle selector tends to be buried at the end of a very long argument list, this comment helps to call out what's going on, but if it's preferred not to add anything extraneous to the generated shaders, I'd be fine removing it.

Reply:
We sometimes add these sort of comments to help debugging - it should be fine.